File: objc/objcfind/Tests/objcfindTests/FileResultTests.m
#import <XCTest/XCTest.h>
#import "FileTypes.h"
#import "FileResult.h"

@interface FileResultTests : XCTestCase
//@property FileTypes *fileTypes;
@end

@implementation FileResultTests

- (void)testFileResultAbsPath {
    NSString *path = @"/Users/cary/src/xfind/objc/objcfind/objcfind/FileResult.m";
    FileResult *fileResult = [[FileResult alloc]
                              initWithFilePath:path
                              fileType:FileTypeCode
                              stat:nil];
    NSString *fileResultString = [fileResult description];
    XCTAssert([fileResultString isEqualToString:path]);
}

- (void)testFileResultTildePath {
    NSString *path = @"~/src/xfind/objc/objcfind/objcfind/FileResult.m";
    FileResult *fileResult = [[FileResult alloc]
                              initWithFilePath:path
                              fileType:FileTypeCode
                              stat:nil];
    NSString *fileResultString = [fileResult description];
    XCTAssert([fileResultString isEqualToString:path]);
}

- (void)testFileResultRelPath1 {
    NSString *path = @"./FileResult.m";
    FileResult *fileResult = [[FileResult alloc]
                              initWithFilePath:path
                              fileType:FileTypeCode
                              stat:nil];
    NSString *fileResultString = [fileResult description];
    XCTAssert([fileResultString isEqualToString:path]);
}

- (void)testFileResultRelPath2 {
    NSString *path = @"../FileResult.m";
    FileResult *fileResult = [[FileResult alloc]
                              initWithFilePath:path
                              fileType:FileTypeCode
                              stat:nil];
    NSString *fileResultString = [fileResult description];
    XCTAssert([fileResultString isEqualToString:path]);
}

@end

File: swift/swiftfind/Tests/swiftfindTests/FileResultTests.swift
//
//  FileResultTests.swift
//  swiftfindTests
//
//  Created by Cary Clark on 5/18/15.
//  Copyright (c) 2015 Cary Clark. All rights reserved.
//

import XCTest

import swiftfind

class FileResultTests: XCTestCase {

    func testFileResultAbsPath() {
        let filePath = "\(NSHomeDirectory())/src/xfind/swift/swiftfind/Sources/swiftfind/FileResult.swift"
        let fileResult = FileResult(filePath: filePath, fileType: FileType.code)
        XCTAssertEqual(filePath, fileResult.description)
    }

    func testFileResultTildePath() {
        let filePath = "~/src/xfind/swift/swiftfind/Sources/swiftfind/FileResult.swift"
        let fileResult = FileResult(filePath: filePath, fileType: FileType.code)
        XCTAssertEqual(filePath, fileResult.description)
    }

    func testFileResultRelPath1() {
        let filePath = "./FileResult.swift"
        let fileResult = FileResult(filePath: filePath, fileType: FileType.code)
        XCTAssertEqual(filePath, fileResult.description)
    }

    func testFileResultRelPath2() {
        let filePath = "../FileResult.swift"
        let fileResult = FileResult(filePath: filePath, fileType: FileType.code)
        XCTAssertEqual(filePath, fileResult.description)
    }

    static var allTests = [
        ("testFileResultAbsPath", testFileResultAbsPath),
        ("testFileResultTildePath", testFileResultTildePath),
        ("testFileResultRelPath1", testFileResultRelPath1),
        ("testFileResultRelPath2", testFileResultRelPath2)
    ]
}
